scrapy/downloadermiddlewares/robotstxt.py

```diff
@@ -94,7 +95,9 @@ def _parse_robots(self, response, netloc):
                 # Running rp.parse() will set rp state from
                 # 'disallow all' to 'allow any'.
                 pass
-        rp.parse(body.splitlines())
+        # stdlib's robotparser expects native 'str' ;
+        # with unicode input, non-ASCII encoded bytes decoding fails in Python2
+        rp.parse(to_native_str(body).splitlines())
```

Comment:
Nice! 

This uses utf-8 in Python 2, even if original page encoding is different, right? Is this correct? Likely it is, but I don't understand the fix well enough.